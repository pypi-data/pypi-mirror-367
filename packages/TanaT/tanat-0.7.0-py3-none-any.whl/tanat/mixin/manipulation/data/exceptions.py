#!/usr/bin/env python3
"""
Exceptions for data access mixins.
"""

from ....exception import TanatException


class DataAccessError(TanatException):
    """Base class for data access exceptions."""


class SequenceDataError(DataAccessError):
    """Exception raised for sequence data access errors."""


class StaticDataError(DataAccessError):
    """Exception raised for static data access errors."""
