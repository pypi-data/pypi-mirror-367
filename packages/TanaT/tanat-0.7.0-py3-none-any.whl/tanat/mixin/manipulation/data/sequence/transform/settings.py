#!/usr/bin/env python3
"""
Transform settings for sequence objects.
"""

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer


@viewer
@dataclass
class TransformSettings:
    """
    Settings for data transformation.

    Attributes:
        relative_time_column: Name of the column created for relative time values.
            Default: "__RELATIVE_TIME__"
        relative_rank_column: Name of the column created for relative rank values.
            Default: "__RELATIVE_RANK__"
        occurrence_column: Name of the column created for occurrence values.
            Default: "__OCCURRENCE__"
        time_spent_column: Name of the column created for time spent values.
            Default: "__TIME_SPENT__"
        frequency_column: Name of the column created for frequency values.
            Default: "__FREQUENCY__"
        time_proportion_column: Name of the column created for time proportion values.
            Default: "__TIME_PROPORTION__"
        distribution_column: Name of the column created for distribution values.
            Default: "__DISTRIBUTION__"
    """

    relative_time_column: str = "__RELATIVE_TIME__"
    relative_rank_column: str = "__RELATIVE_RANK__"
    occurrence_column: str = "__OCCURRENCE__"
    time_spent_column: str = "__TIME_SPENT__"
    frequency_column: str = "__FREQUENCY__"
    time_proportion_column: str = "__TIME_PROPORTION__"
    distribution_column: str = "__DISTRIBUTION__"
