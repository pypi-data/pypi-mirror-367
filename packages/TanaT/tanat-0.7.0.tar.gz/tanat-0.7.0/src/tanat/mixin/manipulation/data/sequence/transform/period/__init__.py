#!/usr/bin/env python3
"""Period data transformer module."""

from .base import PeriodDataTransformer

__all__ = ["PeriodDataTransformer"]
