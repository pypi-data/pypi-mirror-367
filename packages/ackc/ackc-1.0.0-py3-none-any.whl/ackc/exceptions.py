"""Custom exceptions.
"""


class ClientError(Exception):
    """Base exception for all client errors."""
    pass


class AuthError(ClientError):
    """Authentication/authorization error."""
    pass


class TokenExpiredError(AuthError):
    """Access token has expired."""
    pass


class InvalidTokenError(AuthError):
    """Token is invalid or malformed."""
    pass


class UserNotFoundError(ClientError):
    """User not found in Keycloak."""
    pass


class RealmNotFoundError(ClientError):
    """Realm not found in Keycloak."""
    pass


class ClientNotFoundError(ClientError):
    """Client not found in Keycloak."""
    pass


class APIError(ClientError):
    """General API operation error."""
    pass
