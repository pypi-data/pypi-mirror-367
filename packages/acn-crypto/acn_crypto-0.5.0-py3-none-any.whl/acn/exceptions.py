class ACNError(Exception):
    pass

class KeyStructureError(ACNError):
    pass

class ConfigurationError(ACNError):
    pass

class DataError(ACNError):
    pass