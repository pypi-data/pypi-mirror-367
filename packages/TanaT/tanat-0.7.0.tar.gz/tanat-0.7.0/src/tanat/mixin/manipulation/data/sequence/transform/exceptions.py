#!/usr/bin/env python3
"""
Exceptions for sequence data transformation.
"""

from ...exceptions import SequenceDataError


class TransformationError(SequenceDataError):
    """Exception raised for errors in sequence data transformation."""
