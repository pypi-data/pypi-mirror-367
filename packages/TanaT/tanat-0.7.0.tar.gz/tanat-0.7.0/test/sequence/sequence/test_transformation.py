#!/usr/bin/env python3
"""
Test sequence transformations on unique sequence.
"""

import pytest

from tanat.mixin.manipulation.data.sequence.transform.exceptions import (
    TransformationError,
)


class TestTransformationMethod:
    """
    Tests for the transformation methods used on unique sequences.
    """

    @pytest.mark.parametrize("seq_type", ["event", "state", "interval"])
    def test_to_relative_time_tzero_default(self, sequence_pools, seq_type, snapshot):
        """
        Transform a sequence to relative time with default T zero.
        """
        # Get a single sequence from the pool
        sequence = sequence_pools[seq_type][3]
        # T zero is defaulted to the first position in the sequence
        transform_df = sequence.to_relative_time(granularity="day")

        snapshot.assert_match(transform_df.to_csv())

    @pytest.mark.parametrize("seq_type", ["event", "state", "interval"])
    def test_to_relative_rank_tzero_default(self, sequence_pools, seq_type, snapshot):
        """
        Transform a sequence to relative rank with default T zero.
        """
        # Get a single sequence from the pool
        sequence = sequence_pools[seq_type][3]
        # T zero is defaulted to the first position in the sequence
        transform_df = sequence.to_relative_rank()

        snapshot.assert_match(transform_df.to_csv())

    @pytest.mark.parametrize("seq_type", ["event", "state", "interval"])
    def test_to_relative_time_tzero_modified(self, sequence_pools, seq_type, snapshot):
        """
        Transform a sequence to relative time with modified T zero.
        """
        # Get a single sequence from the pool
        sequence = sequence_pools[seq_type][3]
        # T zero is modified to be the second position in the sequence
        sequence.zero_from_position(2)
        transform_df = sequence.to_relative_time(granularity="day")

        snapshot.assert_match(transform_df.to_csv())

    @pytest.mark.parametrize("seq_type", ["event", "state", "interval"])
    def test_to_relative_rank_tzero_modified(self, sequence_pools, seq_type, snapshot):
        """
        Transform a sequence to relative rank with modified T zero.
        """
        # Get a single sequence from the pool
        sequence = sequence_pools[seq_type][3]
        # T zero is modified to be the second position in the sequence
        sequence.zero_from_position(2)
        transform_df = sequence.to_relative_rank()

        snapshot.assert_match(transform_df.to_csv())

    @pytest.mark.parametrize("seq_type", ["event", "state", "interval"])
    def test_to_relative_time_granularity_modified(
        self, sequence_pools, seq_type, snapshot
    ):
        """
        Transform a sequence to relative time with modified granularity.
        """
        # Get a single sequence from the pool
        sequence = sequence_pools[seq_type][3]
        # T zero is defaulted to the first position in the sequence
        transform_df = sequence.to_relative_time(granularity="hour")

        snapshot.assert_match(transform_df.to_csv())

    def test_zero_from_position_out_of_bounds_raises(self, sequence_pools):
        """
        Setting zero_from_position with an out-of-bounds index should raise
        TransformationError on transformation.
        """
        sequence = sequence_pools["event"][3]
        sequence.zero_from_position(2000)

        with pytest.raises(TransformationError):
            sequence.to_relative_time(granularity="day")

        ## Same for relative rank
        with pytest.raises(TransformationError):
            sequence.to_relative_rank()

    @pytest.mark.parametrize("seq_type", ["event", "state", "interval"])
    @pytest.mark.parametrize("by_id", [False, True])
    def test_to_occurrence(self, sequence_pools, seq_type, by_id, snapshot):
        """
        Test the to_occurrence transformation on a single sequence with by_id True/False.
        """
        sequence = sequence_pools[seq_type][3]
        transform_df = sequence.to_occurrence(by_id=by_id, drop_na=True)

        snapshot.assert_match(transform_df.to_csv())

    @pytest.mark.parametrize("seq_type", ["event", "state", "interval"])
    @pytest.mark.parametrize("by_id", [False, True])
    def test_to_occurrence_frequency(self, sequence_pools, seq_type, by_id, snapshot):
        """
        Test the to_occurrence_frequency transformation on a single sequence with by_id True/False.
        """
        sequence = sequence_pools[seq_type][3]
        transform_df = sequence.to_occurrence_frequency(by_id=by_id, drop_na=True)

        snapshot.assert_match(transform_df.to_csv())

    @pytest.mark.parametrize("seq_type", ["event", "state", "interval"])
    @pytest.mark.parametrize("by_id", [False, True])
    def test_to_time_spent(self, sequence_pools, seq_type, by_id, snapshot):
        """
        Test the to_time_spent transformation on a single sequence with by_id True/False.
        """
        sequence = sequence_pools[seq_type][3]
        transform_df = sequence.to_time_spent(
            by_id=by_id, granularity="day", drop_na=True
        )

        snapshot.assert_match(transform_df.to_csv())

    @pytest.mark.parametrize(
        "seq_type,entity_features",
        [
            ("event", ["provider", "FAKE_FEATURES"]),  # FAKE FEATURES should be skipped
            ("state", ["health_state"]),
            ("interval", ["medication", "administration_route"]),
        ],
    )
    @pytest.mark.parametrize("by_id", [False, True])
    def test_to_time_spent_with_entity_features(
        self, sequence_pools, seq_type, entity_features, by_id, snapshot
    ):
        """
        Test the to_time_spent transformation with entity features and by_id True/False on a single sequence.
        """
        sequence = sequence_pools[seq_type][3]
        transform_df = sequence.to_time_spent(
            by_id=by_id,
            granularity="day",
            drop_na=True,
            entity_features=entity_features,
        )

        snapshot.assert_match(transform_df.to_csv())

    @pytest.mark.parametrize("seq_type", ["event", "state", "interval"])
    @pytest.mark.parametrize("by_id", [False, True])
    def test_to_time_proportion(self, sequence_pools, seq_type, by_id, snapshot):
        """
        Test the to_time_proportion transformation on a single sequence with by_id True/False.
        """
        sequence = sequence_pools[seq_type][3]
        transform_df = sequence.to_time_spent(
            by_id=by_id, granularity="day", proportion=True, drop_na=True
        )

        snapshot.assert_match(transform_df.to_csv())
