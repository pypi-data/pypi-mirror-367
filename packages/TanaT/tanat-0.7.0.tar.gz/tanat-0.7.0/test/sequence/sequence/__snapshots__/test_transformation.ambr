# serializer version: 1
# name: TestTransformationMethod.test_to_occurrence[False-event]
  '''
  ,event_type,provider,__OCCURRENCE__
  0,CONSULTATION,RADIOLOGIST,2
  1,EMERGENCY,NURSE,2
  2,FOLLOW_UP,NURSE,2
  3,EMERGENCY,LABORATORY,1
  4,EXAM,GENERAL_PRACTITIONER,1
  5,FOLLOW_UP,LABORATORY,1
  6,FOLLOW_UP,SPECIALIST,1
  7,VACCINATION,RADIOLOGIST,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence[False-interval]
  '''
  ,medication,administration_route,dosage,__OCCURRENCE__
  0,ANTIBIOTIC,INHALATION,3,1
  1,ANTIHYPERTENSIVE,ORAL,2,1
  2,PAIN_RELIEVER,ORAL,2,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence[False-state]
  '''
  ,health_state,condition,__OCCURRENCE__
  0,CHRONIC_MONITORING,NEUROLOGICAL,1
  1,CONVALESCENCE,CARDIAC,1
  2,REMISSION,CARDIAC,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence[True-event]
  '''
  ,patient_id,event_type,provider,__OCCURRENCE__
  0,3,CONSULTATION,RADIOLOGIST,2
  1,3,EMERGENCY,NURSE,2
  2,3,FOLLOW_UP,NURSE,2
  3,3,EMERGENCY,LABORATORY,1
  4,3,EXAM,GENERAL_PRACTITIONER,1
  5,3,FOLLOW_UP,LABORATORY,1
  6,3,FOLLOW_UP,SPECIALIST,1
  7,3,VACCINATION,RADIOLOGIST,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence[True-interval]
  '''
  ,patient_id,medication,administration_route,dosage,__OCCURRENCE__
  0,3,ANTIBIOTIC,INHALATION,3,1
  1,3,ANTIHYPERTENSIVE,ORAL,2,1
  2,3,PAIN_RELIEVER,ORAL,2,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence[True-state]
  '''
  ,patient_id,health_state,condition,__OCCURRENCE__
  0,3,CHRONIC_MONITORING,NEUROLOGICAL,1
  1,3,CONVALESCENCE,CARDIAC,1
  2,3,REMISSION,CARDIAC,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence_frequency[False-event]
  '''
  ,event_type,provider,__FREQUENCY__
  0,CONSULTATION,RADIOLOGIST,0.18181818181818182
  1,EMERGENCY,NURSE,0.18181818181818182
  2,FOLLOW_UP,NURSE,0.18181818181818182
  3,EMERGENCY,LABORATORY,0.09090909090909091
  4,EXAM,GENERAL_PRACTITIONER,0.09090909090909091
  5,FOLLOW_UP,LABORATORY,0.09090909090909091
  6,FOLLOW_UP,SPECIALIST,0.09090909090909091
  7,VACCINATION,RADIOLOGIST,0.09090909090909091
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence_frequency[False-interval]
  '''
  ,medication,administration_route,dosage,__FREQUENCY__
  0,ANTIBIOTIC,INHALATION,3,0.3333333333333333
  1,ANTIHYPERTENSIVE,ORAL,2,0.3333333333333333
  2,PAIN_RELIEVER,ORAL,2,0.3333333333333333
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence_frequency[False-state]
  '''
  ,health_state,condition,__FREQUENCY__
  0,CHRONIC_MONITORING,NEUROLOGICAL,0.3333333333333333
  1,CONVALESCENCE,CARDIAC,0.3333333333333333
  2,REMISSION,CARDIAC,0.3333333333333333
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence_frequency[True-event]
  '''
  ,patient_id,event_type,provider,__FREQUENCY__
  0,3,CONSULTATION,RADIOLOGIST,0.18181818181818182
  1,3,EMERGENCY,NURSE,0.18181818181818182
  2,3,FOLLOW_UP,NURSE,0.18181818181818182
  3,3,EMERGENCY,LABORATORY,0.09090909090909091
  4,3,EXAM,GENERAL_PRACTITIONER,0.09090909090909091
  5,3,FOLLOW_UP,LABORATORY,0.09090909090909091
  6,3,FOLLOW_UP,SPECIALIST,0.09090909090909091
  7,3,VACCINATION,RADIOLOGIST,0.09090909090909091
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence_frequency[True-interval]
  '''
  ,patient_id,medication,administration_route,dosage,__FREQUENCY__
  0,3,ANTIBIOTIC,INHALATION,3,0.3333333333333333
  1,3,ANTIHYPERTENSIVE,ORAL,2,0.3333333333333333
  2,3,PAIN_RELIEVER,ORAL,2,0.3333333333333333
  
  '''
# ---
# name: TestTransformationMethod.test_to_occurrence_frequency[True-state]
  '''
  ,patient_id,health_state,condition,__FREQUENCY__
  0,3,CHRONIC_MONITORING,NEUROLOGICAL,0.3333333333333333
  1,3,CONVALESCENCE,CARDIAC,0.3333333333333333
  2,3,REMISSION,CARDIAC,0.3333333333333333
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_rank_tzero_default[event]
  '''
  patient_id,event_type,provider,__RELATIVE_RANK__
  3,CONSULTATION,RADIOLOGIST,0
  3,FOLLOW_UP,NURSE,1
  3,FOLLOW_UP,LABORATORY,2
  3,CONSULTATION,,3
  3,EMERGENCY,,4
  3,EMERGENCY,NURSE,5
  3,EMERGENCY,NURSE,6
  3,FOLLOW_UP,NURSE,7
  3,EXAM,GENERAL_PRACTITIONER,8
  3,VACCINATION,RADIOLOGIST,9
  3,FOLLOW_UP,SPECIALIST,10
  3,EMERGENCY,LABORATORY,11
  3,CONSULTATION,RADIOLOGIST,12
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_rank_tzero_default[interval]
  '''
  patient_id,medication,administration_route,dosage,__RELATIVE_RANK__
  3,ANTIBIOTIC,INHALATION,3,
  3,PAIN_RELIEVER,ORAL,2,
  3,ANTIHYPERTENSIVE,ORAL,2,
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_rank_tzero_default[state]
  '''
  patient_id,health_state,condition,__RELATIVE_RANK__
  3,CONVALESCENCE,CARDIAC,
  3,CHRONIC_MONITORING,NEUROLOGICAL,
  3,REMISSION,CARDIAC,
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_rank_tzero_modified[event]
  '''
  patient_id,event_type,provider,__RELATIVE_RANK__
  3,CONSULTATION,RADIOLOGIST,-2
  3,FOLLOW_UP,NURSE,-1
  3,FOLLOW_UP,LABORATORY,0
  3,CONSULTATION,,1
  3,EMERGENCY,,2
  3,EMERGENCY,NURSE,3
  3,EMERGENCY,NURSE,4
  3,FOLLOW_UP,NURSE,5
  3,EXAM,GENERAL_PRACTITIONER,6
  3,VACCINATION,RADIOLOGIST,7
  3,FOLLOW_UP,SPECIALIST,8
  3,EMERGENCY,LABORATORY,9
  3,CONSULTATION,RADIOLOGIST,10
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_rank_tzero_modified[interval]
  '''
  patient_id,medication,administration_route,dosage,__RELATIVE_RANK__
  3,ANTIBIOTIC,INHALATION,3,-2
  3,PAIN_RELIEVER,ORAL,2,-1
  3,ANTIHYPERTENSIVE,ORAL,2,0
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_rank_tzero_modified[state]
  '''
  patient_id,health_state,condition,__RELATIVE_RANK__
  3,CONVALESCENCE,CARDIAC,-2
  3,CHRONIC_MONITORING,NEUROLOGICAL,-1
  3,REMISSION,CARDIAC,0
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_time_granularity_modified[event]
  '''
  patient_id,date,event_type,provider
  3,0.0,CONSULTATION,RADIOLOGIST
  3,120.0,FOLLOW_UP,NURSE
  3,312.0,FOLLOW_UP,LABORATORY
  3,888.0,CONSULTATION,
  3,1272.0,EMERGENCY,
  3,2592.0,EMERGENCY,NURSE
  3,3456.0,EMERGENCY,NURSE
  3,4776.0,FOLLOW_UP,NURSE
  3,4872.0,EXAM,GENERAL_PRACTITIONER
  3,5376.0,VACCINATION,RADIOLOGIST
  3,5496.0,FOLLOW_UP,SPECIALIST
  3,6264.0,EMERGENCY,LABORATORY
  3,7584.0,CONSULTATION,RADIOLOGIST
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_time_granularity_modified[interval]
  '''
  patient_id,start_date,end_date,medication,administration_route,dosage
  3,,,ANTIBIOTIC,INHALATION,3
  3,,,PAIN_RELIEVER,ORAL,2
  3,,,ANTIHYPERTENSIVE,ORAL,2
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_time_granularity_modified[state]
  '''
  patient_id,start_date,end_date,health_state,condition
  3,,,CONVALESCENCE,CARDIAC
  3,,,CHRONIC_MONITORING,NEUROLOGICAL
  3,,,REMISSION,CARDIAC
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_time_tzero_default[event]
  '''
  patient_id,date,event_type,provider
  3,0.0,CONSULTATION,RADIOLOGIST
  3,5.0,FOLLOW_UP,NURSE
  3,13.0,FOLLOW_UP,LABORATORY
  3,37.0,CONSULTATION,
  3,53.0,EMERGENCY,
  3,108.0,EMERGENCY,NURSE
  3,144.0,EMERGENCY,NURSE
  3,199.0,FOLLOW_UP,NURSE
  3,203.0,EXAM,GENERAL_PRACTITIONER
  3,224.0,VACCINATION,RADIOLOGIST
  3,229.0,FOLLOW_UP,SPECIALIST
  3,261.0,EMERGENCY,LABORATORY
  3,316.0,CONSULTATION,RADIOLOGIST
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_time_tzero_default[interval]
  '''
  patient_id,start_date,end_date,medication,administration_route,dosage
  3,,,ANTIBIOTIC,INHALATION,3
  3,,,PAIN_RELIEVER,ORAL,2
  3,,,ANTIHYPERTENSIVE,ORAL,2
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_time_tzero_default[state]
  '''
  patient_id,start_date,end_date,health_state,condition
  3,,,CONVALESCENCE,CARDIAC
  3,,,CHRONIC_MONITORING,NEUROLOGICAL
  3,,,REMISSION,CARDIAC
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_time_tzero_modified[event]
  '''
  patient_id,date,event_type,provider
  3,-13.0,CONSULTATION,RADIOLOGIST
  3,-8.0,FOLLOW_UP,NURSE
  3,0.0,FOLLOW_UP,LABORATORY
  3,24.0,CONSULTATION,
  3,40.0,EMERGENCY,
  3,95.0,EMERGENCY,NURSE
  3,131.0,EMERGENCY,NURSE
  3,186.0,FOLLOW_UP,NURSE
  3,190.0,EXAM,GENERAL_PRACTITIONER
  3,211.0,VACCINATION,RADIOLOGIST
  3,216.0,FOLLOW_UP,SPECIALIST
  3,248.0,EMERGENCY,LABORATORY
  3,303.0,CONSULTATION,RADIOLOGIST
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_time_tzero_modified[interval]
  '''
  patient_id,start_date,end_date,medication,administration_route,dosage
  3,-178.0,-169.0,ANTIBIOTIC,INHALATION,3
  3,-16.0,11.0,PAIN_RELIEVER,ORAL,2
  3,0.0,23.0,ANTIHYPERTENSIVE,ORAL,2
  
  '''
# ---
# name: TestTransformationMethod.test_to_relative_time_tzero_modified[state]
  '''
  patient_id,start_date,end_date,health_state,condition
  3,-348.0,-254.0,CONVALESCENCE,CARDIAC
  3,-254.0,0.0,CHRONIC_MONITORING,NEUROLOGICAL
  3,0.0,16.0,REMISSION,CARDIAC
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_proportion[False-event]
  '''
  ,event_type,provider,__TIME_PROPORTION__
  0,CONSULTATION,RADIOLOGIST,0.18181818181818182
  1,EMERGENCY,NURSE,0.18181818181818182
  2,FOLLOW_UP,NURSE,0.18181818181818182
  3,EMERGENCY,LABORATORY,0.09090909090909091
  4,EXAM,GENERAL_PRACTITIONER,0.09090909090909091
  5,FOLLOW_UP,LABORATORY,0.09090909090909091
  6,FOLLOW_UP,SPECIALIST,0.09090909090909091
  7,VACCINATION,RADIOLOGIST,0.09090909090909091
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_proportion[False-interval]
  '''
  ,medication,administration_route,dosage,__TIME_PROPORTION__
  0,ANTIBIOTIC,INHALATION,3,0.15254237288135594
  1,ANTIHYPERTENSIVE,ORAL,2,0.3898305084745763
  2,PAIN_RELIEVER,ORAL,2,0.4576271186440678
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_proportion[False-state]
  '''
  ,health_state,condition,__TIME_PROPORTION__
  0,CHRONIC_MONITORING,NEUROLOGICAL,0.6978021978021978
  1,CONVALESCENCE,CARDIAC,0.25824175824175827
  2,REMISSION,CARDIAC,0.04395604395604396
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_proportion[True-event]
  '''
  ,patient_id,event_type,provider,__TIME_PROPORTION__
  0,3,CONSULTATION,RADIOLOGIST,0.18181818181818182
  1,3,EMERGENCY,NURSE,0.18181818181818182
  2,3,FOLLOW_UP,NURSE,0.18181818181818182
  3,3,EMERGENCY,LABORATORY,0.09090909090909091
  4,3,EXAM,GENERAL_PRACTITIONER,0.09090909090909091
  5,3,FOLLOW_UP,LABORATORY,0.09090909090909091
  6,3,FOLLOW_UP,SPECIALIST,0.09090909090909091
  7,3,VACCINATION,RADIOLOGIST,0.09090909090909091
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_proportion[True-interval]
  '''
  ,patient_id,medication,administration_route,dosage,__TIME_PROPORTION__
  0,3,ANTIBIOTIC,INHALATION,3,0.15254237288135594
  1,3,ANTIHYPERTENSIVE,ORAL,2,0.3898305084745763
  2,3,PAIN_RELIEVER,ORAL,2,0.4576271186440678
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_proportion[True-state]
  '''
  ,patient_id,health_state,condition,__TIME_PROPORTION__
  0,3,CHRONIC_MONITORING,NEUROLOGICAL,0.6978021978021978
  1,3,CONVALESCENCE,CARDIAC,0.25824175824175827
  2,3,REMISSION,CARDIAC,0.04395604395604396
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent[False-event]
  '''
  ,event_type,provider,__TIME_SPENT__
  0,CONSULTATION,RADIOLOGIST,2
  1,EMERGENCY,NURSE,2
  2,FOLLOW_UP,NURSE,2
  3,EMERGENCY,LABORATORY,1
  4,EXAM,GENERAL_PRACTITIONER,1
  5,FOLLOW_UP,LABORATORY,1
  6,FOLLOW_UP,SPECIALIST,1
  7,VACCINATION,RADIOLOGIST,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent[False-interval]
  '''
  ,medication,administration_route,dosage,__TIME_SPENT__
  0,ANTIBIOTIC,INHALATION,3,9.0
  1,ANTIHYPERTENSIVE,ORAL,2,23.0
  2,PAIN_RELIEVER,ORAL,2,27.0
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent[False-state]
  '''
  ,health_state,condition,__TIME_SPENT__
  0,CHRONIC_MONITORING,NEUROLOGICAL,254.0
  1,CONVALESCENCE,CARDIAC,94.0
  2,REMISSION,CARDIAC,16.0
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent[True-event]
  '''
  ,patient_id,event_type,provider,__TIME_SPENT__
  0,3,CONSULTATION,RADIOLOGIST,2
  1,3,EMERGENCY,NURSE,2
  2,3,FOLLOW_UP,NURSE,2
  3,3,EMERGENCY,LABORATORY,1
  4,3,EXAM,GENERAL_PRACTITIONER,1
  5,3,FOLLOW_UP,LABORATORY,1
  6,3,FOLLOW_UP,SPECIALIST,1
  7,3,VACCINATION,RADIOLOGIST,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent[True-interval]
  '''
  ,patient_id,medication,administration_route,dosage,__TIME_SPENT__
  0,3,ANTIBIOTIC,INHALATION,3,9.0
  1,3,ANTIHYPERTENSIVE,ORAL,2,23.0
  2,3,PAIN_RELIEVER,ORAL,2,27.0
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent[True-state]
  '''
  ,patient_id,health_state,condition,__TIME_SPENT__
  0,3,CHRONIC_MONITORING,NEUROLOGICAL,254.0
  1,3,CONVALESCENCE,CARDIAC,94.0
  2,3,REMISSION,CARDIAC,16.0
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent_with_entity_features[False-event-entity_features0]
  '''
  ,provider,__TIME_SPENT__
  0,NURSE,4
  1,RADIOLOGIST,3
  2,LABORATORY,2
  3,GENERAL_PRACTITIONER,1
  4,SPECIALIST,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent_with_entity_features[False-interval-entity_features2]
  '''
  ,medication,administration_route,__TIME_SPENT__
  0,ANTIBIOTIC,INHALATION,9.0
  1,ANTIHYPERTENSIVE,ORAL,23.0
  2,PAIN_RELIEVER,ORAL,27.0
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent_with_entity_features[False-state-entity_features1]
  '''
  ,health_state,__TIME_SPENT__
  0,CHRONIC_MONITORING,254.0
  1,CONVALESCENCE,94.0
  2,REMISSION,16.0
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent_with_entity_features[True-event-entity_features0]
  '''
  ,patient_id,provider,__TIME_SPENT__
  0,3,NURSE,4
  1,3,RADIOLOGIST,3
  2,3,LABORATORY,2
  3,3,GENERAL_PRACTITIONER,1
  4,3,SPECIALIST,1
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent_with_entity_features[True-interval-entity_features2]
  '''
  ,patient_id,medication,administration_route,__TIME_SPENT__
  0,3,ANTIBIOTIC,INHALATION,9.0
  1,3,ANTIHYPERTENSIVE,ORAL,23.0
  2,3,PAIN_RELIEVER,ORAL,27.0
  
  '''
# ---
# name: TestTransformationMethod.test_to_time_spent_with_entity_features[True-state-entity_features1]
  '''
  ,patient_id,health_state,__TIME_SPENT__
  0,3,CHRONIC_MONITORING,254.0
  1,3,CONVALESCENCE,94.0
  2,3,REMISSION,16.0
  
  '''
# ---
